### PR TITLE
Feat/registry

### LINKED ISSUE #22: GeneratorRegistry
Implement a registry containing all types of Generator implementations. 

Requires at least one of #10, #11, #12, #13, #14, #15

### LINKED ISSUE #23: ControllerRegistry
Implement a registry containing all types of Controller implementations. 

Requires at least one of #17, #18, #19, #20, #21

### LINKED ISSUE #79: generic Registry class
transfer old Registry to be a subclass of Registry (DatasetRegistry)

### PR DESCRIPTION
**Pull request by tediiieee**
- Implemented abstract Registry class
- Changed already existing DatasetRegistry to be a subclass. 
- Implemented subclasses ControllerRegistry and GeneratorRegistry
- Refactored files to use hardcoded constants instead of random strings

Closes #79
Closes #23 
Closes #22

### CODE REVIEW

File: src/dataset/DatasetRegistry.py [hunk 1]
```diff
@@ -1,24 +1,18 @@
...
-    """
-
-    _DATASETS = None
 
-    @staticmethod
-    def init_registry():
-        """
-        Static method to initialize `DATASETS` with Dataset subclasses.
-        Calls the constructor for each dataset with predefined parameters.
+    * If more Datasets are implemented they should be added in this function.
```
Comment (sporreking) [inline src/dataset/DatasetRegistry.py:12]: What function?

"If more Datasets are implemented they must be manually added to the internal storage of this class." perhaps

File: src/generator/GeneratorRegistry.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+class GeneratorRegistry(Registry):
+    """
+    Static class implementing the abstract Registry class
+    used for initialization and storing of all subclasses of the
+    Generator class.
+    """
+
+    # * If more Generators are implemented they should be added in this dict.
```
Comment (sporreking) [inline src/generator/GeneratorRegistry.py:14]: This format would fix the comment in `DatasetRegistry`!

**Review by sporreking — CHANGES_REQUESTED** [review]
Everything looks good, except for a few small things I noticed.

One thing I thought about is that maybe the get_names(), get_resources(), and get_resource() could be implemented generally, since they seem to require a strict format of nestled access etc. (forcing all subclasses to implement something specific may be redundant). This would most likely be a later issue however.

File: src/core/Registry.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    def get_names() -> list[Union[str, Tuple]]:
+        """
+        Returns all the names (keys) in the registry. In the case of
+        nestled resource storing structure, the multiple keys are stored in
+        tuples of strings instead of string.
+
+        Returns:
+            list[Union[str, Tuple]]: List of all names (keys) in the registry. Tuple if
+                nestled resource storing structure, e.g., [("ASAD", "unetgan"),...]
+                such that resource can be achieved from storing struct ´d´ d["ASAD"]["unetgan"].
```
[the remark this answers is not in the document]
Comment (tediiieee) [reply src/core/Registry.py:22]: Accessed is correct. Restated comment such that it reflects how the functions are used, not how the internal structures work. (`get_resource(get_names()[0])` should work)

**Review by sporreking — APPROVED** [review]
Godlike!

**Pull request merged**